client/checkout/platform-checkout/email-input-iframe.js

```diff
@@ -285,7 +285,9 @@ export const handlePlatformCheckoutEmailInput = ( field, api ) => {
 				wcpayTracks.recordUserEvent(
 					wcpayTracks.events.PLATFORM_CHECKOUT_OTP_COMPLETE
 				);
-				api.initPlatformCheckout().then( ( response ) => {
+				api.initPlatformCheckout(
+					e.data.platformCheckoutUserSession
+				).then( ( response ) => {
```

Comment:
One thing that is not a part of this PR but we may want to handle now, is adding a catch-block in case something goes wrong with the `api.initPlatformCheckout(...)` call. Something like the following:
```javascript
api.initPlatformCheckout( e.data.platformCheckoutUserSession )
	.then( ( response ) => {
		if ( 'success' === response.result ) {
			window.location = response.url;
		} else {
			closeIframe();
		}
	} )
	.catch( ( error ) => {
		// Handle error.
	}
```

This way, things don't silently fail.

Reply:
I created #4157 for this. I talked to David in p1650326845599849-slack-C022WMN88KG and thought we could address that in a different PR. Thanks for raising this.